dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/LdapConnection.java

```diff
@@ -2114,27 +2122,36 @@ private void supportRangeAttributes(Attributes attributes, String dn, TimedDirCo
                         }
                     }
                     Attribute result = results.get(attributeWithRange);
+                    //If the result is null, we didn't find the exact attributeWithRange string
+                    //we were looking for. This could mean there are fewer results left than iAttrRangeStep.
+                    //In this case, LDAP will have sent back range;x-*.
                     if (result == null) {
-                        Object[] args = {
-                                          Integer.valueOf(rangeLow).toString()
+                        Object[] rangeArgs = {
+                                               Integer.toString(rangeLow)
                         };
-                        attributeWithRange = attrId + MessageFormat.format(ATTR_RANGE_LAST_QUERY, args);
+                        String tempRange = attributeWithRange;
+                        attributeWithRange = attrId + MessageFormat.format(ATTR_RANGE_LAST_QUERY, rangeArgs);
+                        if (tc.isDebugEnabled()) {
+                            Tr.debug(tc,
+                                     "Result was null for " + tempRange + ", now parsing results for " + attributeWithRange);
+                        }
                         result = results.get(attributeWithRange);
-                        lastQuery = true;
+                        //If we found a result, this is the last search.
+                        if (result != null)
+                            quitLoop = true;
                     }
+                    //If we found a result, add it to the list. If the result was range;x-*, quit.
+                    //If result is null, there was an issue parsing the results. Quit the loop.
                     if (result != null) {
                         for (NamingEnumeration<?> neu2 = result.getAll(); neu2.hasMoreElements();) {
                             newAttr.add(neu2.nextElement());
                         }
 
-                        if (lastQuery) {
-                            quitLoop = true;
-                        } else {
-                            rangeLow = rangeHigh + 1;
-                            rangeHigh = rangeLow + iAttrRangeStep - 1;
-                        }
+                        rangeLow = rangeHigh + 1;
+                        rangeHigh = rangeLow + localAttrRangeStep - 1;
                     } else {
-                        lastQuery = true;
+                        quitLoop = true;
+                        Tr.error(tc, WIMMessageKey.RANGE_ATTRIBUTE_PARSING, dn, localAttrRangeStep);
```

Comment:
Should this be warning? It doesn't seem to be fatal.

Reply:
Or do we stop processing completely? Do we get partial or no results?

Reply:
As discussed over Slack, this may mean partial results are being returned, so we should throw an exception.